tests/integration/override_targets/BUILD

```diff
@@ -69,3 +69,20 @@ sh_test(
         "@bazel_tools//tools/bash/runfiles",
     ],
 )
+
+genquery(
+    name = "root_module_can_override",
+    expression = "deps(@root_module_can_override//:com_squareup_okhttp3_okhttp)",
+    opts = [
+        "--nohost_deps",
+        "--noimplicit_deps",
+    ],
+    scope = ["@root_module_can_override//:com_squareup_okhttp3_okhttp"],
+)
+
+sh_test(
```

Comment:
Elsewhere we've used a `diff_test` to check this kind of thing. There's an example of how to ignore the test when not using `bzlmod` [here](https://github.com/bazel-contrib/rules_jvm_external/pull/1373/files#diff-b7f7f72456bab30f131cdd91dd097e0f9555b53963ad19c635b5693687621022R119-R125) It would be nice to continue using the same approach here, if only for consistency in our testing.

Reply:
Makes sense.
This would require refactoring the OG tests added by @jin 
Will push the changes by tomorrow.